### PR TITLE
feature(thorntail): Add Thorntail generators

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

File: catalog/generators/platform-thorntail/info.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "type": "generator",
+  "name": "platform thorntail",
```
Comment (quintesse) [inline catalog/generators/platform-thorntail/info.json]: Could we make this more like the Vert.x one? "Thorntail Base Platform"

File: catalog/generators/database-crud-thorntail/files/src/main/fabric8/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                   key: user
+            - name: DB_PASSWORD
+              valueFrom:
+                 secretKeyRef:
+                   name: my-database-secret
+                   key: password
+            - name: GC_MAX_METASPACE_SIZE
+              value: "150"
+            - name: JAVA_OPTIONS
+              value: "-Dswarm.datasources.data-sources.MyDS.connection-url=jdbc:postgresql://$(MY_DATABASE_SERVICE_HOST):$(MY_DATABASE_SERVICE_PORT)/my_data -Dswarm.datasources.data-sources.MyDS.user-name=$(DB_USERNAME) -Dswarm.datasources.data-sources.MyDS.password=$(DB_PASSWORD) -Dswarm.datasources.data-sources.MyDS.driver-name=postgresql"
```
Comment (quintesse) [inline catalog/generators/database-crud-thorntail/files/src/main/fabric8/deployment.yml]: The `MY_DATABASE_SERVICE_HOST` still needs to be changed, right?

Comment (gastaldi) [reply catalog/generators/database-crud-thorntail/files/src/main/fabric8/deployment.yml]: This file doesn't exist anymore

File: catalog/generators/platform-thorntail/info.json (hunk 1 above)
Comment (gastaldi) [reply catalog/generators/platform-thorntail/info.json]: I copied from the node.js one, sure

**Comment by edewit** [comment]
I get the following:
```
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'profiles.profile.id' must be unique but found duplicate profile with id openshift @ line 158, column 11
[ERROR] 'profiles.profile.id' must be unique but found duplicate profile with id openshift-it @ line 181, column 11
[ERROR] 'profiles.profile.id' must be unique but found duplicate profile with id local @ line 210, column 11
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.openshift.appgen:my-app:1.0.0 (/tmp/bla/pom.xml) has 3 errors
[ERROR]     'profiles.profile.id' must be unique but found duplicate profile with id openshift @ line 158, column 11
[ERROR]     'profiles.profile.id' must be unique but found duplicate profile with id openshift-it @ line 181, column 11
[ERROR]     'profiles.profile.id' must be unique but found duplicate profile with id local @ line 210, column 11
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```
This is due to the fact that the profile `local` is in the pom twice

Steps to reproduce:
```
$> yarn -s apply /tmp/bla "bla" thorntail database '{ "databaseType": "postgresql" }'
$> cd /tmp/bla
$> mvn install
```

**Comment by gastaldi** [comment]
@edewit it is fixed in the latest maven-model-interpreter JAR. You need to run `yarn install` first (maybe with a force option to replace the JAR inside `lib/core/maven`)

File: catalog/capabilities/database/index.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import DatabasePostgresql from 'generators/database-postgresql';
 import DatabaseMysql from 'generators/database-mysql';
 import DatabaseCrudVertx from 'generators/database-crud-vertx';
 import DatabaseCrudNodejs from 'generators/database-crud-nodejs';
+import DatabaseCrudThorntail from "generators/database-crud-thorntail";
```
[the remark this answers is not in the document]
Comment (gastaldi) [reply catalog/capabilities/database/index.ts]: Thanks for the hint, I'll have a look at that

**Comment by quintesse** [comment]
Yeah, you'll have to run `yarn install --force`

**Pull request merged**